TYPECHECKER_TODO.md

```diff
@@ -31,22 +31,22 @@ Sprint 2:
 - [x] Traverse AST first time to get declarations and bind to scope
 
 - [ ] Run typechecker for RaiseStmt, ensure string
-- [ ] Implement parser for type signatures
-- [ ] Run and implement type checker and reasoning for FnStmt
+- [x] Implement parser for type signatures
+- [x] Run and implement type checker and reasoning for FnStmt
 - [ ] Record, in the AST, the positions of the symbols, in order to give better error messages
-- [ ] Run typechecker and see type rules for bluerprints, traits, structs and impls
-- [ ] Assert the context of ReturnStmt and pass the expected return type to it when inside functions
+- [ ] Run typechecker and see type rules for traits, structs and impls
+- [/] Assert the context of ReturnStmt and pass the expected return type to it when inside functions
```

Comment:
No (for bodied-functions), only for expression-functions, so this is something like a half. It is partially done.